test/Core.Test/Auth/Models/Business/Tokenables/SsoEmail2faSessionTokenableTests.cs

```diff
@@ -0,0 +1,173 @@
+﻿using AutoFixture.Xunit2;
+using Bit.Core.Auth.Models.Business.Tokenables;
+using Bit.Core.Entities;
+using Bit.Core.Tokens;
+using Xunit;
+namespace Bit.Core.Test.Auth.Models.Business.Tokenables;
+
+// Note: these test names follow MethodName_StateUnderTest_ExpectedBehavior pattern.
+public class SsoEmail2faSessionTokenableTests
+{
+    // Allow a small tolerance for possible execution delays or clock precision to avoid flaky tests.
+    private readonly TimeSpan _timeTolerance = TimeSpan.FromMilliseconds(10);
+
+    /// <summary>
+    /// Tests the default constructor behavior when passed a null user.
+    /// </summary>
+    [Fact]
+    public void Constructor_NullUser_PropertiesSetToDefault()
+    {
+        var token = new SsoEmail2faSessionTokenable(null);
+
+        Assert.Equal(default, token.Id);
+        Assert.Equal(default, token.Email);
+    }
+
+    /// <summary>
+    /// Tests that when a valid user is provided to the constructor, the resulting token properties match the user.
+    /// </summary>
+    [Theory, AutoData]
+    public void Constructor_ValidUser_PropertiesSetFromUser(User user)
+    {
+        var token = new SsoEmail2faSessionTokenable(user);
+
+        Assert.Equal(user.Id, token.Id);
+        Assert.Equal(user.Email, token.Email);
+    }
+
+    /// <summary>
+    /// Tests the default expiration behavior immediately after initialization.
+    /// </summary>
+    [Fact]
+    public void Constructor_AfterInitialization_ExpirationSetToExpectedDuration()
+    {
+        var token = new SsoEmail2faSessionTokenable();
+        var expectedExpiration = DateTime.UtcNow + SsoEmail2faSessionTokenable.GetTokenLifetime();
+
+        Assert.True(expectedExpiration - token.ExpirationDate < _timeTolerance);
+    }
+
+    /// <summary>
+    /// Tests that a custom expiration date is preserved after token initialization.
+    /// </summary>
+    [Fact]
+    public void Constructor_CustomExpirationDate_ExpirationMatchesProvidedValue()
+    {
+        var customExpiration = DateTime.UtcNow.AddHours(3);
+        var token = new SsoEmail2faSessionTokenable
+        {
+            ExpirationDate = customExpiration
+        };
+
+        Assert.True(Math.Abs((customExpiration - token.ExpirationDate).TotalMilliseconds) < _timeTolerance.TotalMilliseconds);
```

Comment:
I believe you can instead do: `(customExpiration - token.ExpirationDate).Duration() < _timeTolerance` to me at least, that feels more readable.

Reply:
Ah, that is much cleaner! Thank you!